File: src/main/java/shopping/member/infrastructure/persistence/jpa/MemberRepositoryImpl.java
package shopping.member.infrastructure.persistence.jpa;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;
import shopping.member.domain.MemberEntity;
import shopping.member.domain.MemberRepository;

import java.util.Optional;

@Repository
@RequiredArgsConstructor
public class MemberRepositoryImpl implements MemberRepository {

    private final MemberJpaRepository memberJpaRepository;

    @Override
    public MemberEntity save(MemberEntity member) {
        return memberJpaRepository.save(member);
    }

    @Override
    public Optional<MemberEntity> findByEmail(String email) {
        return memberJpaRepository.findByEmail(email);
    }
}

File: src/main/kotlin/shopping/infrastructure/member/MemberRepositoryImpl.kt
package shopping.infrastructure.member

import org.springframework.stereotype.Component
import shopping.domain.member.Member
import shopping.domain.member.MemberRepository

@Component
class MemberRepositoryImpl(
    private val memberJpaRepository: MemberJpaRepository,
) : MemberRepository {
    override fun findByEmail(email: String): Member? = memberJpaRepository.findByEmail(email)

    override fun existsByEmail(email: String): Boolean = memberJpaRepository.existsByEmail(email)

    override fun save(member: Member): Member = memberJpaRepository.save(member)
}
